pkg/tsdb/cloudwatch/client_factory.go

```diff
@@ -49,10 +49,10 @@ var NewCWLogsClient = func(sess *session.Session) cloudwatchlogsiface.CloudWatch
 	return cloudwatchlogs.New(sess)
 }
 
-// EC2 client factory.
+// NewEC2Client is a client factory.
 //
 // Stubbable by tests.
-var newEC2Client = func(provider client.ConfigProvider) models.EC2APIProvider {
+var NewEC2Client = func(provider client.ConfigProvider) models.EC2APIProvider {
```

Comment:
How come the signature isn't `func(sess *session.Session) models.EC2APIProvider` like the other refactors?

Reply:
I realized we didn't need to change this to get it to work so I reverted it. When I look at the documentation they describe the first arg as a clientProvider so I kept it, although in the example they pass a session? https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#New I guess sessions implement clientProviders? Felt like if I didn't have to change this I shouldn't, but open to your thoughts!

Reply:
Can't think off the top of my mind why we had it the other way. Now it seems to be a mix. I see below it's this way, too. Doesn't seem like a big deal. Thanks for having a look!